src/undo.h

```diff
@@ -59,31 +92,177 @@ class CTxUndo
 public:
     // undo information for all txins
     std::vector<CTxInUndo> vprevout;
+    uint256 refTx;            //hash of coins from ceased sidechains. It's not needed for ordinary coins and certs
```

Comment:
instead of using CTxUndo for both certs/txs and ceased sidechain certificate btws removal, restore the previous CTxUndo for regular certs/txs and introduce another CVoidedCertificateUndo used in case of ceased sidechain. CVoidedCertificateUndo  will contain voidedCertHash and firstBwtPos without the need of marker for backward compatibility. CBlockUndo will keep a vector of CVoidedCertificateUndo.